doc improvement : applications: nrf_desktop: Document forced scan state (#12745)
nrfconnect/sdk-nrf: applications/nrf_desktop/doc/ble_scan.rst

The scanning is never interrupted if there is no connected Bluetooth peer.
If :ref:`CONFIG_DESKTOP_BLE_SCAN_PM_EVENTS <config_desktop_app_options>` is enabled, the power down event will also interrupt scanning.

Forced scan
-----------

The module supports a dedicated forced scan state that prevents interrupting scanning when a connected peripheral is in use.
The forced scan speeds up establishing new connections with peripherals, but it also negatively impacts performance of already connected peripherals.
If the :ref:`CONFIG_DESKTOP_BLE_FORCED_SCAN_DURATION_S <config_desktop_app_options>` Kconfig option is set to a value greater than zero, the module enters the forced scan state after one of the following conditions:

* boot or wakeup,
* successful peripheral discovery,
* peripheral disconnection,
* connection failure,
* bond erase,
* scan request.

> After the time specified by the :ref:`CONFIG_DESKTOP_BLE_FORCED_SCAN_DURATION_S <config_desktop_app_options>`, the module switches to regular scanning that can be interrupted by using connected peripherals and times out after the scan duration specified by :ref:`CONFIG_DESKTOP_BLE_SCAN_DURATION_S <config_desktop_app_options>` if there are peripherals connected over Bluetooth.

Reviewer: I think we can break the sentence into two parts so it is simpler. Here is my suggestion 

```suggestion
After the time specified by the :ref:`CONFIG_DESKTOP_BLE_FORCED_SCAN_DURATION_S <config_desktop_app_options>`, the module switches to regular scanning that can be interrupted by using connected peripherals.
The operation times out after the scan duration specified by :ref:`CONFIG_DESKTOP_BLE_SCAN_DURATION_S <config_desktop_app_options>` if there are peripherals connected over Bluetooth.
```

Author: Rephrased.

> After the time specified by the :ref:`CONFIG_DESKTOP_BLE_FORCED_SCAN_DURATION_S <config_desktop_app_options>`, the module switches to regular scanning that can be interrupted by using connected peripherals and times out after the scan duration specified by :ref:`CONFIG_DESKTOP_BLE_SCAN_DURATION_S <config_desktop_app_options>` if there are peripherals connected over Bluetooth.

Reviewer 2: ```suggestion
After the duration that is specified in the :ref:`CONFIG_DESKTOP_BLE_FORCED_SCAN_DURATION_S <config_desktop_app_options>` Kconfig option, the module switches to regular scanning.
The regular scanning can be interrupted if there are peripherals connected over Bluetooth and times out after the scan duration specified in the :ref:`CONFIG_DESKTOP_BLE_SCAN_DURATION_S <config_desktop_app_options>`.
```
Hope the meaning has not changed.
